src/lib.rs

```diff
@@ -147,22 +147,22 @@ impl<K: Ord, V> IntervalTree<K, V> {
     /// Queries the interval tree for all elements overlapping a given interval.
     ///
     /// This runs in `O(log n + m)`.
-    pub fn query(&self, range: Range<K>) -> QueryIter<K, V> {
+    pub fn query(&self, range: Range<K>) -> QueryIter<K, V, Range<K>> {
         QueryIter {
             todo: self.todo(),
             tree: self,
-            query: Query::Range(range),
+            query: range,
         }
     }
 
     /// Queries the interval tree for all elements containing a given point.
     ///
     /// This runs in `O(log n + m)`.
-    pub fn query_point(&self, point: K) -> QueryIter<K, V> {
+    pub fn query_point(&self, point: K) -> QueryIter<K, V, K> {
```

Comment:
If we would want to, we could now also unify these two methods into

```rust
pub fn query<Q: Query<K>>(&self, query: Q) -> QueryIter<K, V, Q>;
```

working for both ranges and points but forcing the `Query` trait to be public.